api/v1alpha1/ext_proc_types.go

```diff
@@ -0,0 +1,29 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// +kubebuilder:validation:XValidation:rule="has(self.service) ? (!has(self.service.backendRef.group) || self.service.backendRef.group == \"\") : true", message="group is invalid, only the core API group (specified by omitting the group field or setting it to an empty string) is supported"
+// +kubebuilder:validation:XValidation:rule="has(self.service) ? (!has(self.service.backendRef.kind) || self.service.backendRef.kind == 'Service') : true", message="kind is invalid, only Service (specified by omitting the kind field or setting it to 'Service') is supported"
+//
+// ExtProc defines the configuration for External Processing filter.
+type ExtProc struct {
```

Comment:
We may need an `insertBefor` property in this struct to specify the place where this ExtProc extension should be inserted in the filter chain.

https://github.com/envoyproxy/gateway/blob/dbaa9a0ad8c35ac78909de30814cc63b08e0e2c1/api/v1alpha1/wasm_types.go#L42

Reply:
Right. I think that this can be tackled in several iterations:

1. Extensions have a built-in location (e.g. after auth)
2. Support "global" filter ordering through: #2571
3. Support GW/Route specific filter ordering through a `InsertBefore`, `stage` or local `filterOrder`. 

For the first iteration, I propose to keep this as-is, and go for (3) when requirements come up.

Reply:
Thought again, we just need a `Priority` inside the Extension struct to allow users to adjust the location of individual extensions.

Reply:
within the list, we can always use the list order as the inherent order (ascending or descending)

Reply:
Yes, we can use the list order for the priority between extensions, but we would also need an implicit priority field to decide the location of an extension in other filters(jwt, cors, oauth2,etc.)